src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -134,6 +134,39 @@ public override bool Equals(object obj)
     /// </summary>
     public abstract partial class WorkspaceModel : NotificationObject, ILocatable, IUndoRedoRecorderClient, ILogSource, IDisposable, IWorkspaceModel
     {
+        #region nested classes
+
+        /// <summary>
+        /// This class enables the delay of graph execution.
+        /// Use instances of this class to specify a code scope in which you want graph execution to be delayed. 
+        /// Class is thread safe, although behavior is not well defined. 
+        /// Nested instance of this class do not have a well defined behavior.
+        /// </summary>
+        internal class DelayedGraphExecution : IDisposable
+        {
+            private readonly WorkspaceModel workspace;
+            private static readonly object stateMutex = new object();
+
+            public DelayedGraphExecution(WorkspaceModel wModel)
+            {
+                workspace = wModel;
+                lock (stateMutex)
+                {
+                    workspace.delayGraphExecutionCounter++;
+                }
+            }
+
+            public virtual void Dispose()
+            {
+                lock (stateMutex)
+                {
+                    workspace.delayGraphExecutionCounter--;
+                }
+                workspace.RequestRun();
```

Comment:
There is a `RequestRun` call I think, at the end of each of these actions - `undo, redo, convertnodetocode`, so why call it again here?

Reply:
`DelayedGraphExecution` can be used in other cases, besides the ones I covered in this PR (that already call RequestRun at the end)
The `DelayedGraphExecution` class does not know that there will be a RequestRun called after it is used.
In those cases of `undo, redo, convertnodetocode etc` the requestRun calls (after `DelayedGraphExecution`) will not register any UpdateGraphAsyncTasks, because there will be no more dirty nodes.